### PR TITLE
Tobago 1589

### PR DESCRIPTION
**Pull request by bohmber**
Fix for the mojarra issue.
Created the components transient and move decode to sheet back.
The explicit remove of the component is needed for mojarra 2.3 (they have the DynamicActionComponent List inside the state handling. Looks like it's an other workaround for some crap and they did not respect the transient property)

### CODE REVIEW

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java [hunk 1]
```diff
@@ -132,13 +139,86 @@ public void decode(final FacesContext facesContext, final UIComponent component)
...
+    decodeSheetAction(facesContext, sheet);
+    decodeColumnAction(facesContext, columns);
 /* this will be done by the javax.faces.component.UIData.processDecodes() because these are facets.
     for (UIComponent facet : sheet.getFacets().values()) {
       facet.decode(facesContext);
     }
 */
   }
 
+  private void decodeColumnAction(final FacesContext facesContext, List<AbstractUIColumnBase> columns) {
```
Comment (lofwyr14) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java]: please use final as parameter qualifier

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java [hunk 2]
```diff
@@ -132,13 +139,86 @@ public void decode(final FacesContext facesContext, final UIComponent component)
...
+        if (sorterId.equals(sourceId)) {
+          final UIData data = (UIData) column.getParent();
+          data.queueEvent(new SortActionEvent(data, column));
+        }
+      }
+    }
+  }
+
+
+  private void decodeSheetAction(final FacesContext facesContext, UISheet component) {
```
Comment (lofwyr14) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java]: please use final as parameter qualifier

File: tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java [hunk 3]
```diff
@@ -132,13 +139,86 @@ public void decode(final FacesContext facesContext, final UIComponent component)
...
+    final String sourceId = facesContext.getExternalContext().getRequestParameterMap().get("javax.faces.source");
+
+    final String clientId = component.getClientId(facesContext);
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("sourceId = '{}'", sourceId);
+      LOG.debug("clientId = '{}'", clientId);
+    }
+
+    final String sheetClientIdWithAction = clientId + UINamingContainer.getSeparatorChar(facesContext) + SUFFIX_PAGE_ACTION;
```
Comment (lofwyr14) [inline tobago-core/src/main/java/org/apache/myfaces/tobago/internal/renderkit/renderer/SheetRenderer.java]: line width > 120 (please run checkstyle)

**Comment by bohmber** [comment]
Checkstyle errors are fixed now.

**Comment by bohmber** [comment]
Added more fixes for Mojarra. Mojarra is like Klendathu

**Comment by bohmber** [comment]
There is still an issue with mojarra 2.0 PostAddToViewEvent, Head and StateHandling in mojarra 2.0. No more energy to find a workaround for the crap mojarra impl.

**Pull request merged**